src/GafferSceneUI/LightTool.cpp

```diff
@@ -0,0 +1,1266 @@
+	IntVectorDataPtr vertIdsData = new IntVectorData;
+	V3fVectorDataPtr pData = new V3fVectorData;
+
+	std::vector<int> &vertsPerPoly = vertsPerPolyData->writable();
+	std::vector<int> &vertIds = vertIdsData->writable();
+	std::vector<V3f> &p = pData->writable();
+
+	const int numCircleDivisions = 100;
+	const int numSegments = std::max( 1, (int)ceil( abs( stopFraction - startFraction ) * numCircleDivisions ) );
+
+	p.push_back( V3f( 0, 0, radius ) );
+
+	for( int i = 0; i < numSegments + 1; ++i )
+	{
+		const float a = ( startFraction + ( stopFraction - startFraction ) * (float)i / (float)numSegments ) * 2.f * M_PI;
+		p.push_back( V3f( -sin( a ), 0, -cos( a ) ) * radius + V3f( 0, 0, radius ) );
+	}
+
+	for( int i = 0; i < numSegments; ++i )
+	{
+		vertIds.push_back( i + 1 );
+		vertIds.push_back( i + 2 );
+		vertIds.push_back( 0 );
+		vertsPerPoly.push_back( 3 );
+	}
+
+	IECoreScene::MeshPrimitivePtr solidArc = new IECoreScene::MeshPrimitive( vertsPerPolyData, vertIdsData, "linear", pData );
+	solidArc->variables["N"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, new V3fData( V3f( 0, 1, 0 ) ) );
+	solidArc->variables["Cs"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, new Color3fData( color ) );
+
+	return solidArc;
+}
+
+class LightToolHandle : public Handle
+{
+
+	public :
+
+		/// \todo SpotLightTool (below) is not using `metadataKey` because it knows what keys it needs.
+		/// If that is going to be the case for other tools, we should remove it.
+		LightToolHandle(
+			const std::string &attributePattern,
+			const InternedString &metadataKey,
+			const std::string &name
+		) :
+			Handle( name ),
+			m_attributePattern( attributePattern ),
+			m_metadataKey( metadataKey ),
+			m_lookThroughLight( false )
+		{
+
+		}
+		~LightToolHandle()
+		{
+
+		}
+
+		// Update inspectors and other data needed to display and interact with the tool.
+		// Derived classes should call this parent method first, then implement custom logic.
+		virtual void update( ScenePathPtr scenePath, const PlugPtr &editScope )
+		{
+			m_handleScenePath = scenePath;
+			m_editScope = editScope;
+		}
+
+		const std::string attributePattern() const
+		{
+			return m_attributePattern;
+		}
+
+		ScenePath *handleScenePath() const
+		{
+			return m_handleScenePath.get();
+		}
+
+		Plug *editScope() const
+		{
+			return m_editScope.get();
+		}
+
+		void setLookThroughLight( bool lookThroughLight )
+		{
+			m_lookThroughLight = lookThroughLight;
+		}
+
+		bool getLookThroughLight() const
+		{
+			return m_lookThroughLight;
+		}
+
+		// Called at the beginning of a drag with the path to be inspected
+		// set in the current context. Must be implemented by derived classes to collect
+		// whatever data will be needed to implement the drag operation.
+		virtual void addDragInspection() = 0;
+
+		virtual bool handleDragMove( const GafferUI::DragDropEvent &event ) = 0;
+		virtual bool handleDragEnd() = 0;
```

Comment:
What's the reason for these being public methods called by LightTool, rather than protected methods called by LightToolHandle?